bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableRegionLocator.java

```diff
@@ -56,11 +57,16 @@
   private long regionsFetchTimeMillis;
 
   public AbstractBigtableRegionLocator(
-      TableName tableName, BigtableOptions options, IBigtableDataClient client) {
+      TableName tableName, BigtableHBaseSettings settings, IBigtableDataClient client) {
     this.tableName = tableName;
     this.client = client;
-    this.bigtableTableName = options.getInstanceName().toTableName(tableName.getNameAsString());
-    ServerName serverName = ServerName.valueOf(options.getDataHost(), options.getPort(), 0);
+    this.bigtableTableName =
+        new BigtableTableName(
```

Comment:
how frequently do we need to do this operation? Is it worth adding this as a getter on `BigtableHbaseSettings` (similar to the convenience method `toTableName()`)?

Reply:
This is an intermediate state, We would be able to replace `BigtableTableName` with string `tableID` once the veneer client wrappers are in place.

Reply:
sounds good, thanks!